File: backend/Service/src/main/java/ru/nikidzawa/db/store/client/dto/ChatRoomDto.java
package ru.nikidzawa.db.store.client.dto;

import lombok.*;
import lombok.experimental.FieldDefaults;
import ru.nikidzawa.db.store.entity.IndividualChatEntity;
import ru.nikidzawa.db.store.entity.IndividualEntity;

import java.util.List;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@FieldDefaults(level = AccessLevel.PRIVATE)
public class ChatRoomDto {
    IndividualDtoShort companion;
    List<MessageDto> messages;
    IndividualChatEntity chat;
}

File: frontend/react-app/src/api/internal/dto/ChatRoomDto.js
import IndividualDtoShort from "./IndividualDtoShort";
import MessageDto from "./MessageDto";
import PrivateChatDto from "./PrivateChatDto";

export default class ChatRoomDto {
    constructor(companion, messages, chat, unreadCount) {
        this.companion = companion;
        this.messages = messages;
        this.chat = chat;
        this.unreadCount = unreadCount;

        this.firstLoadMessages = true;
    }

    static fromJSON(data) {
        return new ChatRoomDto(
            IndividualDtoShort.fromJSON(data.companion),
            [MessageDto.fromJSON(data.lastMessage)],
            PrivateChatDto.fromJSON(data.chat),
            data.unreadCount
        );
    }
}
